### PR TITLE
Updated the README.md with prerequisites required for running the tests.

### PR DESCRIPTION
**Pull request by Senjuti256**
# Changes

I have updated the README.md in the test folder with some prerequisites required for running the tests.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -22,33 +10,35 @@ go test $(go list ./... | grep -v third_party/)
...
+
+2. Next install Tekton Triggers in the cluster. Depending on which version of Triggers you want to install run one of the commands from [Installation](https://github.com/tektoncd/triggers/blob/main/docs/install.md#installation)
+
+Set environment variable `TEST_CLUSTERTASK_LIST_EMPTY` to any value if tests are run in an environment which contains `clustertasks`. Set `SYSTEM_NAMESPACE` variable to appropriate value (say tekton-pipelines).The `SYSTEM_NAMESPACE` variable provides a configurable way to specify the namespace where Tekton Pipelines system components reside within the Kubernetes cluster.
+
+
+### Running
+
+End to end tests live in this directory. By default `go test` will not run [the end to end tests](#end-to-end-tests),
+which need `-tags=e2e` to be enabled hence to run these tests, you must provide `go` with `-tags=e2e`.
```
Comment (vinamra28) [inline test/README.md:27]:
```suggestion
it need `-tags=e2e` to be enabled in order to run these tests, hence you must provide `go` with `-tags=e2e`. 
```

File: test/README.md [hunk 2]
```diff
@@ -22,33 +10,35 @@ go test $(go list ./... | grep -v third_party/)
...
+1. To run end to end tests, you will need to have a running Tekton pipeline deployed on your cluster, see [Install Pipeline](../DEVELOPMENT.md#install-pipeline).
+
+2. Next install Tekton Triggers in the cluster. Depending on which version of Triggers you want to install run one of the commands from [Installation](https://github.com/tektoncd/triggers/blob/main/docs/install.md#installation)
+
+Set environment variable `TEST_CLUSTERTASK_LIST_EMPTY` to any value if tests are run in an environment which contains `clustertasks`. Set `SYSTEM_NAMESPACE` variable to appropriate value (say tekton-pipelines).The `SYSTEM_NAMESPACE` variable provides a configurable way to specify the namespace where Tekton Pipelines system components reside within the Kubernetes cluster.
+
+
+### Running
+
+End to end tests live in this directory. By default `go test` will not run [the end to end tests](#end-to-end-tests),
```
Comment (vinamra28) [inline test/README.md:26]: >End to end tests live in this directory.

let's add a hyperlink to `this` keyword ?

File: test/README.md [hunk 3]
```diff
@@ -74,10 +64,21 @@ go test -v -count=1 ./test/e2e/... --kubeconfig ~/special/kubeconfig --cluster m
...
 
+### Prerequisite to run make check and make generated 
+
+Ensure that golangci-lint and yamllint are installed on your development machine. If not then install them following [golangci-lint](https://golangci-lint.run/usage/install/) and [yamllint](https://yamllint.readthedocs.io/en/stable/quickstart.html).
+
+Next step will be to run
+```bash
+# For linting using golangci-lint
+make check
+# For running tests to update goldenfile
```
Comment (vinamra28) [inline test/README.md:82]:
```suggestion
# For running tests to check diff in goldenfile and [docs](https://github.com/tektoncd/cli/tree/main/docs)
```

File: test/README.md (hunk 2 above)
Comment (Senjuti256) [reply test/README.md:26]: Sure @vinamra28 doing it.

**Review by vinamra28 — COMMENTED** [review]
@Senjuti256, can you please squash the commits?
/lgtm

**Pull request merged**